File: Scala+/ScalaSbt/Libs+/Databricks+/DbUtils/src/test/scala/dbutils/DbUtilsIT.scala
package dbutils

import com.databricks.dbutils_v1.DBUtilsHolder
import org.scalatest.flatspec.AnyFlatSpec
import org.scalatest.matchers.should.Matchers

/**
 * Can work only in Databricks.
 */
class DbUtilsIT extends AnyFlatSpec with Matchers {
  private val dbfsTmpDir = "dbfs:/tmp"
  private val dbutils = DBUtilsHolder.dbutils

  it should "make a directory" in {
    val created = dbutils.fs.mkdirs(dbfsTmpDir + "/dir1/dir2")
    created shouldBe true
  }

  it should "list files in a directory" in {
    val files = dbutils.fs.ls(dbfsTmpDir)
    files.foreach(file => println(file.path))
    files shouldNot be(empty)
  }

}

File: Java+/BigData+/Databricks+/DbUtils/test/dbutils/DbUtilsIT.java
package dbutils;

import com.databricks.backend.daemon.dbutils.FileInfo;
import com.databricks.dbutils_v1.DBUtilsHolder;
import com.databricks.dbutils_v1.DbfsUtils;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * Can work only in Databricks.
 */
class DbUtilsIT {
    private final String dbfsTmpDir = "dbfs:/tmp/" + DbUtilsIT.class.getSimpleName();
    private final DbfsUtils fs = DBUtilsHolder.dbutils().fs();

    @Test
    void listFile() {
        var files = fs.ls(dbfsTmpDir);
        files.foreach(FileInfo::path);
        assertThat(files.size()).isPositive();
    }

    @Test
    void createDirs() {
        var dir = dbfsTmpDir + "/dir1";
        var created = fs.mkdirs(dir);
        assertThat(created).isTrue();
    }

    @Test
    void deleteDirs() {
        var dir = dbfsTmpDir + "/dir1";
        var created = fs.rm(dir, true);
        assertThat(created).isTrue();
    }

}